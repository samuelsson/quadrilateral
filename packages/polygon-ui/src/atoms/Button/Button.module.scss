[data-theme="light"] {
  --Button-background-color: var(--color-blue-gray-800);
  --Button-color: var(--color-white);
}

[data-theme="dark"] {
  --Button-background-color: var(--color-blue-gray-100);
  --Button-color: var(--color-blue-gray-800);
}

.Button {
  height: 48px;
  border: none;
  padding: 0 2rem;
  border-radius: 5px;
  font-weight: bold;
  background-color: var(--Button-background-color);
  color: var(--Button-color);
  text-decoration: none;
  text-transform: uppercase;

  transition-property: background, border, color, box-shadow;
  transition-timing-function: linear;
  transition-duration: 0.2s;

  &:hover:not(:disabled),
  &:focus:not(:disabled),
  &:active:not(:disabled) {
    cursor: pointer;
  }
}
