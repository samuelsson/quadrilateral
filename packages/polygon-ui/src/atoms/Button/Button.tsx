import React from 'react';
import styles from './Button.module.scss';

export interface ButtonProps {
  children: React.ReactNode;
  name: string;
  onClick?: () => void;
  type?: 'button' | 'submit' | 'reset';
  disabled?: boolean;
}

const Button = ({
  children,
  name,
  onClick,
  type,
  disabled,
}: ButtonProps): JSX.Element => (
  <button
    // eslint-disable-next-line react/button-has-type
    type={type || 'button'}
    name={name}
    onClick={onClick}
    disabled={disabled}
    className={styles.Button}
  >
    {children}
  </button>
);

export default Button;
