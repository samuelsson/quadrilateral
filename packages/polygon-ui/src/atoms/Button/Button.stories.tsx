import * as React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';
import { boolean } from '@storybook/addon-knobs';

import Button from './Button';

const ButtonContainer = styled.div`
  button:not(:last-of-type) {
    margin-right: 1rem;
  }
`;

const buttonStyles = ['primary', 'secondary', 'success', 'danger'];

storiesOf('Button', module)
  .addDecorator(storyFn => <ButtonContainer>{storyFn()}</ButtonContainer>)
  .add(
    'Default',
    (): JSX.Element => (
      <>
        {buttonStyles.map(button => (
          <Button
            name={button}
            appearance={button}
            disabled={boolean('Disabled', false)}
          >
            {button.toUpperCase()}
          </Button>
        ))}
      </>
    )
  )
  .add(
    'Bare',
    (): JSX.Element => (
      <>
        {buttonStyles.map(button => (
          <Button
            name={button}
            appearance={button}
            disabled={boolean('Disabled', false)}
            bare
          >
            {button.toUpperCase()}
          </Button>
        ))}
      </>
    )
  );
