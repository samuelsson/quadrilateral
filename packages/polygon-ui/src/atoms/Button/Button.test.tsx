import * as React from 'react';
import { render } from '@testing-library/react';
import Button from './Button';

describe('Button', (): void => {
  it('renders with no errors', (): void => {
    const { asFragment } = render(<Button name="default-button">Hello</Button>);

    expect(asFragment()).toMatchSnapshot();
  });
});
