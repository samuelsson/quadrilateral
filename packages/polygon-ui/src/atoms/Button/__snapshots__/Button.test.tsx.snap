// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with no errors 1`] = `
<DocumentFragment>
  <button
    class="Button"
    name="default-button"
    type="button"
  >
    Hello
  </button>
</DocumentFragment>
`;
